sdk/dotnet/Pulumi.Automation/LocalWorkspace.cs

```diff
@@ -335,6 +353,39 @@ public static Task<WorkspaceStack> CreateOrSelectStackAsync(LocalProgramArgs arg
 
         private static readonly string[] SettingsExtensions = new string[] { ".yaml", ".yml", ".json" };
 
+        private async Task GetPulumiVersionAsync(CancellationToken cancellationToken = default)
+        {
+            var result = await this.RunCommandAsync(new[] { "version" }, cancellationToken).ConfigureAwait(false);
+            var versionString = result.StandardOutput.Trim();
+            if (versionString[0] == 'v') {
+                versionString = versionString.TrimStart('v');
+            }
+            var version = SemVersion.Parse(versionString);
+            if (version == null)
+            {
+                throw new ArgumentNullException("version");
```

Comment:
Throwing an `Argument...` exception here is the same issue as above, since `version` wasn't an argument to this method it seems weird